### PR TITLE
Poster

### PR DESCRIPTION
**Pull request by jakub-drak-sbahi**
image

### CODE REVIEW

File: web/src/app/tree/shared/tree-diagram.service.ts [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+      // see if there's any empty space at the horizontal mid-point in that layer
+      var overlaps = layout.diagram.findObjectsIn(newbnds, function(x) { return x.part; }, function(p) { return p !== v.node; }, true);
+      if (overlaps.count === 0) {
+        v.node.move(newbnds.position);
+      }
+    }
+  });
+};
+
+GenogramLayout.prototype.findParentsMarriageLabelNode = function(node) {
```
Comment (BartoszChrostowski) [inline web/src/app/tree/shared/tree-diagram.service.ts:585]: it isn't genogram 😝

File: web/src/app/tree/shared/tree-diagram.service.ts [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+      callback(reader.result);
+    }
+    reader.readAsDataURL(xhr.response);
+  };
+  xhr.open('GET', url);
+  xhr.responseType = 'blob';
+  xhr.send();
+}
+
+function makeSvg() {
```
Comment (BartoszChrostowski) [inline web/src/app/tree/shared/tree-diagram.service.ts:610]: For later: [PDF](https://gojs.net/latest/projects/pdf/minimalPDF.html`)

**Review by BartoszChrostowski — CHANGES_REQUESTED** [review]

File: web/src/app/tree/shared/tree-diagram.service.ts [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+              $(go.Panel,
+                $(go.Shape,
+                  { stroke: null, strokeWidth: 0 },
+                  new go.Binding("fill", "", attrFill),
+                  new go.Binding("geometry", "", maleGeometry))
+              ),
+            margin: 1
+          },
+          new go.Binding("itemArray", "a")
+        )
```
Comment (BartoszChrostowski) [inline web/src/app/tree/shared/tree-diagram.service.ts:168]: Is this panel used?

**Comment by Dragemil** [comment]
Why there are `null`s on the screenshot?

**Comment by jakub-drak-sbahi** [comment]
> Why there are `null`s on the screenshot?

Because these users haven't provided dates of birth.

File: web/src/app/tree/shared/tree-diagram.service.ts (hunk 1 above)
Comment (jakub-drak-sbahi) [reply web/src/app/tree/shared/tree-diagram.service.ts:585]: ?

Comment (BartoszChrostowski) [reply web/src/app/tree/shared/tree-diagram.service.ts:474]: A genogram is a graphical representation of gene inheritance 👼

Comment (jakub-drak-sbahi) [reply web/src/app/tree/shared/tree-diagram.service.ts:474]: Yes, but it's using genogram code :) Names are original from library author's site.

**Review by BartoszChrostowski — APPROVED** [review]
Ok.

**Pull request merged**